Enable local endpoint testing

hjaltisan opened the pull request:
This adds `--local-endpoint` as a start up parameter option, and if provided, will send analytics to a localhost endpoint instead of the gateway. Could be useful for testing.

BaldvinTh commented on src/libs/installer/eventlogger.cpp:
```diff
@@ -58,7 +58,16 @@ void EventLogger::initialize(eve_launcher::application::Application_Region regio
+
+    if (QInstaller::isLocalEndpointEnabled())
+    {
+        s_gatewayUrl = QString::fromLatin1("https://localhost:5001/weatherforecast");
```
Should this value be configurable via the command line argument? `--local-endpoint=https://...`

hjaltisan replied:
I thought about it, but thought nah, nobody is going to use this, but then again, could be useful, I'll add that.
Only downside of merging this entire PR, is it allows users to bypass sending the data to the gateway. I could always send it to both places 🤔 (just thought that would be overkill)

hjaltisan replied:
Also this would always require us to provide the endpoint (that's just how the Qt implementation works).
Or we could introduce an additional parameter for providing the endpoint url `--local-endpoint-url=https://...`

hjaltisan commented:
@BaldvinTh Updated a bit.

- `--local-endpoint` is now `-t` or `--telemetry-endpoint`
- It now expects a value (`-t=local` or `-t local`)
  - Optional values not supported: https://doc.qt.io/qt-5/qcommandlineparser.html
  - > The parser does not support optional values - if an option is set to require a value, one must be present. If such an option is placed last and has no value, the option will be treated as if it had not been specified.
- Added the following pre defined values:
   - `local` = `https://localhost:5001/weatherforecast`
   - `dev` = *`<ELG_DEV_WORLD_URL>`*
   - `live` = *`<ELG_LIVE_WORLD_URL>`*
   - `cdev` = *`<ELG_DEV_CHINA_URL>`*
   - `clive` = *`<ELG_LIVE_CHINA_URL>`*
   - *`<VALUE_OF_YOUR_CHOICE>`*
- Changed it so it always sends to the correct endpoint, and what the user provides is just sent additionally
   - Unless the user specifies the actual endpoint, in that case the message is only sent once.
- Simplified the `-f`, `--log-to-file` and `--no-automated-log` code a little bit

BaldvinTh reviewed (approved):
Looks good

gretarg commented on src/libs/installer/eventlogger.cpp:
```diff
@@ -101,6 +101,18 @@ void EventLogger::sendAllocatedEvent(google::protobuf::Message* payload)
+    {
+        QString customEndpoint = QInstaller::getProvidedTelemetryEndpoint();
+        qDebug() << "framework | EventLogger::sendAllocatedEvent | Also sending to user provided endpoint =" << customEndpoint;
+        if (customEndpoint == s_gatewayUrl) {
```
Inconsistancy in curlies having/not having their own lines throughout this PR. Don't think we formalized any coding standards (personally I always put them in their own line (don't verify)).

hjaltisan replied:
Yes, I completely agree. But since we don't have any standards (that I'm aware of) I simply followed the format of the code around. But if we prefer brackets having their own lines, then I can easily update the PR
